### ISSUE TITLE
Editor crash in TabContainer::_gui_input

### ISSUE THREAD

**Issue by qarmin** [godotengine/godot#32597]
**Godot version:**
3.2.dev.custom_build. 6d6d43714
**OS/device including version:**
Ubuntu 19.04
**Issue description:**
Backtrace
```
[1] /lib/x86_64-linux-gnu/libc.so.6(+0x43f60) [0x7fcdec090f60] (??:0)
[2] TabContainer::_gui_input(Ref<InputEvent> const&) (/home/rafal/Pulpit/godot/scene/gui/tab_container.cpp:89 (discriminator 3))
[3] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (/home/rafal/Pulpit/godot/./core/method_bind.gen.inc:775 (discriminator 12))
[4] Object::call_multilevel(StringName const&, Variant const**, int) (/home/rafal/Pulpit/godot/core/object.cpp:763 (discriminator 1))
[5] Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/rafal/Pulpit/godot/core/object.cpp:864)
[6] Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) (/home/rafal/Pulpit/godot/scene/main/viewport.cpp:1524 (discriminator 2))
[7] Viewport::_gui_input_event(Ref<InputEvent>) (/home/rafal/Pulpit/godot/scene/main/viewport.cpp:1834 (discriminator 3))
[8] Viewport::input(Ref<InputEvent> const&) (/home/rafal/Pulpit/godot/scene/main/viewport.cpp:2680 (discriminator 2))
[9] Viewport::_vp_input(Ref<InputEvent> const&) (/home/rafal/Pulpit/godot/scene/main/viewport.cpp:1301)
[10] MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) (/home/rafal/Pulpit/godot/./core/method_bind.gen.inc:775 (discriminator 12))
[11] Object::call(StringName const&, Variant const**, int, Variant::CallError&) (/home/rafal/Pulpit/godot/core/object.cpp:921 (discriminator 1))
[12] Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/rafal/Pulpit/godot/core/object.cpp:848)
[13] SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) (/home/rafal/Pulpit/godot/scene/main/scene_tree.cpp:264)
[14] SceneTree::input_event(Ref<InputEvent> const&) (/home/rafal/Pulpit/godot/scene/main/scene_tree.cpp:419 (discriminator 6))
[15] InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) (/home/rafal/Pulpit/godot/main/input_default.cpp:416)
[16] InputDefault::parse_input_event(Ref<InputEvent> const&) (/home/rafal/Pulpit/godot/main/input_default.cpp:260)
[17] InputDefault::flush_accumulated_events() (/home/rafal/Pulpit/godot/main/input_default.cpp:679)
[18] OS_X11::process_xevents() (/home/rafal/Pulpit/godot/platform/x11/os_x11.cpp:2625)
[19] OS_X11::run() (/home/rafal/Pulpit/godot/platform/x11/os_x11.cpp:3188)
[20] godot(main+0x128) [0x13e061a] (/home/rafal/Pulpit/godot/platform/x11/godot_x11.cpp:57)
[21] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xeb) [0x7fcdec073b6b] (??:0)
[22] godot(_start+0x2a) [0x13e043a] (??:?)
```

Address Sanitizer log
```
==10646==ERROR: AddressSanitizer: heap-use-after-free on address 0x61a001447e90 at pc 0x000009f01bf8 bp 0x7ffcf6ecf850 sp 0x7ffcf6ecf840
READ of size 8 at 0x61a001447e90 thread T0
    #0 0x9f01bf7 in TabContainer::_gui_input(Ref<InputEvent> const&) scene/gui/tab_container.cpp:89
    #1 0x2358959 in MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) core/method_bind.gen.inc:775
    #2 0xe5d9137 in Object::call_multilevel(StringName const&, Variant const**, int) core/object.cpp:763
    #3 0xe5dc44e in Object::call_multilevel(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) core/object.cpp:863
    #4 0x9597425 in Viewport::_gui_call_input(Control*, Ref<InputEvent> const&) scene/main/viewport.cpp:1524
    #5 0x95a3b6c in Viewport::_gui_input_event(Ref<InputEvent>) scene/main/viewport.cpp:1834
    #6 0x95cb0d3 in Viewport::input(Ref<InputEvent> const&) scene/main/viewport.cpp:2680
    #7 0x958c0e6 in Viewport::_vp_input(Ref<InputEvent> const&) scene/main/viewport.cpp:1302
    #8 0x2358959 in MethodBind1<Ref<InputEvent> const&>::call(Object*, Variant const**, int, Variant::CallError&) core/method_bind.gen.inc:775
    #9 0xe5dda22 in Object::call(StringName const&, Variant const**, int, Variant::CallError&) core/object.cpp:921
    #10 0xe5dbea2 in Object::call(StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) core/object.cpp:847
    #11 0x947c185 in SceneTree::call_group_flags(unsigned int, StringName const&, StringName const&, Variant const&, Variant const&, Variant const&, Variant const&, Variant const&) scene/main/scene_tree.cpp:264
    #12 0x948168d in SceneTree::input_event(Ref<InputEvent> const&) scene/main/scene_tree.cpp:419
    #13 0x14e2c0a in InputDefault::_parse_input_event_impl(Ref<InputEvent> const&, bool) main/input_default.cpp:442
    #14 0x14d805c in InputDefault::parse_input_event(Ref<InputEvent> const&) main/input_default.cpp:259
    #15 0x14eb22a in InputDefault::flush_accumulated_events() main/input_default.cpp:678
    #16 0x146dada in OS_X11::process_xevents() platform/x11/os_x11.cpp:2624
    #17 0x148061e in OS_X11::run() platform/x11/os_x11.cpp:3188
    #18 0x13fcba3 in main platform/x11/godot_x11.cpp:56
    #19 0x7f5d732acb6a in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x26b6a)
    #20 0x13fc7a9 in _start (/usr/bin/godots+0x13fc7a9)

0x61a001447e90 is located 16 bytes inside of 1256-byte region [0x61a001447e80,0x61a001448368)
freed by thread T0 here:
    #0 0x7f5d74ba804f in __interceptor_free (/lib/x86_64-linux-gnu/libasan.so.5+0x10c04f)
    #1 0xeb7b37f in Memory::free_static(void*, bool) core/os/memory.cpp:181
    #2 0x158da38 in void memdelete<Object>(Object*) core/os/memory.h:118
    #3 0x949a596 in SceneTree::_flush_delete_queue() scene/main/scene_tree.cpp:1110
    #4 0x94887dd in SceneTree::idle(float) scene/main/scene_tree.cpp:534
    #5 0x15831c2 in Main::iteration() main/main.cpp:1978
    #6 0x14807a2 in OS_X11::run() platform/x11/os_x11.cpp:3192
    #7 0x13fcba3 in main platform/x11/godot_x11.cpp:56
    #8 0x7f5d732acb6a in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x26b6a)

previously allocated by thread T0 here:
    #0 0x7f5d74ba8448 in malloc (/lib/x86_64-linux-gnu/libasan.so.5+0x10c448)
    #1 0xeb7a32a in Memory::alloc_static(unsigned long, bool) core/os/memory.cpp:85
    #2 0xeb7a226 in operator new(unsigned long, char const*) core/os/memory.cpp:42
    #3 0x92d0b6b in Object* ClassDB::creator<PopupMenu>() (/usr/bin/godots+0x92d0b6b)
    #4 0xe369acc in ClassDB::instance(StringName const&) core/class_db.cpp:541
    #5 0x1826dc0 in GDScriptNativeClass::instance() modules/gdscript/gdscript.cpp:82
    #6 0x1826669 in GDScriptNativeClass::_new() modules/gdscript/gdscript.cpp:69
    #7 0x18f706f in MethodBind0R<Variant>::call(Object*, Variant const**, int, Variant::CallError&) core/method_bind.gen.inc:237
    #8 0xe5dda22 in Object::call(StringName const&, Variant const**, int, Variant::CallError&) core/object.cpp:921
    #9 0xe8cc578 in Variant::call_ptr(StringName const&, Variant const**, int, Variant*, Variant::CallError&) core/variant_call.cpp:1083
    #10 0x1a157b9 in GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Variant::CallError&, GDScriptFunction::CallState*) modules/gdscript/gdscript_function.cpp:1079
    #11 0x184d376 in GDScriptInstance::call_multilevel(StringName const&, Variant const**, int) modules/gdscript/gdscript.cpp:1180
    #12 0x9366476 in Node::_notification(int) scene/main/node.cpp:60
    #13 0x17068b7 in Node::_notificationv(int, bool) scene/main/node.h:46
    #14 0x1708d09 in CanvasItem::_notificationv(int, bool) scene/2d/canvas_item.h:166
    #15 0x170b3cf in Control::_notificationv(int, bool) scene/gui/control.h:48
    #16 0x1787801 in Container::_notificationv(int, bool) scene/gui/container.h:38
    #17 0x9f3d1f3 in TabContainer::_notificationv(int, bool) scene/gui/tab_container.h:38
    #18 0xe5ddf8d in Object::notification(int, bool) core/object.cpp:931
    #19 0x949666f in SceneTree::_notify_group_pause(StringName const&, int) scene/main/scene_tree.cpp:964
    #20 0x9487208 in SceneTree::idle(float) scene/main/scene_tree.cpp:516
    #21 0x15831c2 in Main::iteration() main/main.cpp:1978
    #22 0x14807a2 in OS_X11::run() platform/x11/os_x11.cpp:3192
    #23 0x13fcba3 in main platform/x11/godot_x11.cpp:56
    #24 0x7f5d732acb6a in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x26b6a)

SUMMARY: AddressSanitizer: heap-use-after-free scene/gui/tab_container.cpp:89 in TabContainer::_gui_input(Ref<InputEvent> const&)
```
**Steps to reproduce**
![jfile](https://user-images.githubusercontent.com/41945903/66272744-f28e4480-e86d-11e9-84a3-9935e2dd75dd.gif)

**Minimal reproduction project:**
[The_worst_Godot_test_project.zip](https://github.com/godotengine/godot/files/3694669/The_worst_Godot_test_project.zip)

Minimal Script:
```
extends TabContainer

func _process(delta) -> void:
	var qq : String = ""
	qq = qq
	
	var popup_menu_temp : PopupMenu = PopupMenu.new()
	if randi() % 2 == 1:
		set_popup(popup_menu_temp)
	if randi() % 2 == 1:
		set_tab_disabled(randi() % 1000 - 500,bool(randi()%2))
	if randi() % 2 == 1:
		set_tab_icon(randi() % 1000 - 500,load("res://RES/Sprite" + str(randi()%4 + 1) + ".png"))
	if randi() % 2 == 1:
		set_tab_title(randi() % 1000 - 500,str(randi() % 1000 - 500))
	if randi() % 2 == 1:
		set_tabs_rearrange_group(randi() % 1000 - 500)
	popup_menu_temp.queue_free()
```

**Comment by girng** [comment]
dang

**Comment by Xrayez** [comment]
Created easy to reproduce minimal project, just hit F5: [tab-container-gui-input-crash.zip](https://github.com/godotengine/godot/files/3698776/tab-container-gui-input-crash.zip) 😛

```gdscript
extends TabContainer

var ev = InputEventMouseButton.new()

func _ready():
	var pm := PopupMenu.new()
	set_popup(pm)
	pm.queue_free()

	yield(get_tree(), "idle_frame")

	ev.pressed = true
	ev.button_index = BUTTON_LEFT
	ev.button_mask = BUTTON_LEFT
	ev.position = Vector2(24, 13)
	Input.parse_input_event(ev)
```
This is how UI can be tested from code!

**Comment by Xrayez** [comment]
Ok, so I'm getting scary crash regarding atomic operations under magical conditions (with the above snippet):
```
/-----------------------------------------
Running Script res://tests/issues/32597/test_tab_container_gui_input.gd
-----------------------------------------/
* test_tab_container_gui_input
    /# Yielding (10s):  Hopefully no crash happens. #/
    /# waiting #/
CrashHandlerException: Program crashed
Dumping the backtrace. Please include this when reporting the bug on https://github.com/godotengine/godot/issues
[0] atomic_decrement (core\safe_refcount.cpp:127)
[1] atomic_decrement (core\safe_refcount.cpp:127)
[2] StringName::unref (core\string_name.cpp:99)
[3] StringName::~StringName (core\string_name.cpp:422)
[4] Reference::~Reference (core\reference.cpp:112)
[5] InputEventMouseButton::`scalar deleting destructor'
[6] Ref<InputEventMouseButton>::unref (core\reference.h:279)
[7] Viewport::_gui_input_event (scene\main\viewport.cpp:2336)
[8] Viewport::input (scene\main\viewport.cpp:2683)
[9] Viewport::_vp_input (scene\main\viewport.cpp:1302)
[10] MethodBind1<Viewport,Ref<InputEvent> const &>::call (core\method_bind.gen.inc:867)
[11] Object::call (core\object.cpp:921)
[12] Object::call (core\object.cpp:848)
[13] SceneTree::call_group_flags (scene\main\scene_tree.cpp:264)
[14] SceneTree::input_event (scene\main\scene_tree.cpp:419)
[15] InputDefault::_parse_input_event_impl (main\input_default.cpp:442)
[16] InputDefault::flush_accumulated_events (main\input_default.cpp:679)
[17] OS_Windows::run (platform\windows\os_windows.cpp:3010)
[18] widechar_main (platform\windows\godot_windows.cpp:162)
[19] _main (platform\windows\godot_windows.cpp:186)
[20] main (platform\windows\godot_windows.cpp:196)
[21] __scrt_common_main_seh (f:\dd\vctools\crt\vcstartup\src\startup\exe_common.inl:288)
[22] BaseThreadInitThunk
-- END OF BACKTRACE --
```
Seems to happen if you **hold** the LMB while clicking on arrow.

Related to #31860?

**Comment by qarmin** [comment]
Fixed in Godot 3.2.2
